.github/workflows/continuous-publish-new-version.yaml

```diff
@@ -0,0 +1,52 @@
+name: Continuous beta version publish to npm
+
+permissions:
+  contents: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - main
+
+# To trigger auto version publish name your squashed commits using conventional notation
+# We support: feat:, fix: and :refactor
+jobs:
+  build:
+    if: ${{ startsWith(github.event.head_commit.message, 'feat:') || startsWith(github.event.head_commit.message, 'fix:') || startsWith(github.event.head_commit.message, 'refactor:') }}
```

Comment:
Just a thought: is this check case sensitive? I know my editor sometimes auto-capitalizes the start of a sentence, and I can see it being very confusing why some commit doesn't trigger the flow just because it starts with "Fix" instead of "fix".

Reply:
feat, fix or refactor after 4.22.0 will be 4.23.0-beta.1.  This is the logic from `npm version prerelease --preid=beta`. I think it makes sense in the continuous world we strive for. If you want to fix something on the official 4.22.0 you should work on a branch off the 4.22.0 tag without blocking people who are already contributing to the next 4.23 official release candidate.

Reply:
Actually two things in one here. Ill just address 4.22.0 -> fix/refactor. After a release is cut on the main branch its normal that the next commits go to the next version. If we have fixes for a patch release I think it makes sense to use a release_branch and cherry pick commits from main that you want in the patch version.

Reply:
the lowercase function from what I know is currently missing in GH actions and if often requested. I prefer to keep the code free of the complex workarounds for now. But we can iterate on this idea next.

Reply:
One more comment about versioning. We could follow traditional conventional commits without alpha/beta and always upgrade major/minor/patch based on conventional commits. this would work if we didn't distribute the package to many clients and only self hosted. Due to the nature of internal releases and open source client hosted releases we make a tradeoff. We continuously bump up the beta version for our internal continuous delivery purposes. But for the client facing stuff we make a manual release on a regular cadence. This GH action only codifies things that we already do manually.